File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int curMin=prices[0], ans=0;
        for (int i:prices) {
            curMin=Math.min(curMin, i);
            ans=Math.max(ans, i-curMin);
        }

        return ans;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.c
int maxProfit(int* prices, int pricesSize) {
    int ans = 0, buy = prices[0];
    for (int i=1; i<pricesSize; ++i) {
        if (prices[i] - buy > ans) {
            ans = prices[i] - buy;
        }

        if (prices[i] < buy) {
            buy = prices[i];
        }
    }

    return ans;
}